criu/kerndat.c

```diff
@@ -1060,19 +1062,51 @@ static int kerndat_has_openat2(void)
 	return 0;
 }
 
-#define KERNDAT_CACHE_FILE     KDAT_RUNDIR "/criu.kdat"
-#define KERNDAT_CACHE_FILE_TMP KDAT_RUNDIR "/.criu.kdat"
+#define KERNDAT_CACHE_NAME	"criu.kdat"
+#define KERNDAT_CACHE_FILE	KDAT_RUNDIR"/"KERNDAT_CACHE_NAME
+
+static int get_kerndat_filename(char **kdat_file)
+{
+	int ret;
+
+	/*
+	 * Running as non-root, even with CAP_CHECKPOINT_RESTORE, does not
+	 * allow to write to KDAT_RUNDIR which usually is only writable by root.
+	 * Let's write .criu.kdat file to the home directory for non-root cases.
+	 */
+	if (opts.uid) {
+		if (!getenv("HOME")) {
```

Comment:
should we store the kdat file in XDG_RUNTIME_DIR?
https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

Reply:
Yes, done in a71f7dc750081a841d6cfe73199019580037a62f.